### PR TITLE
Split unit into three categories 

### PR DESCRIPTION
**Pull request by adespawn**
One category for test that are currently passing because of old datastax code
Second category for tests that are being broken by current changes (this category is not tested on CI)
Third category for unit tests that should work with the new driver code

(Note, CI examples fails due changes in ``simple-query``)

### CODE REVIEW

**Comment by adespawn** [comment]
If needed I can also take ``Splits examples and unit tests into two separate CI tasks.`` from #43 and add it as a separate PR to allow passing of unit tests without working exaples

**Comment by adespawn** [comment]
Although #54 can be easily merged into main, it would require much mor effort to elegantly merge this into main (as this PR already splits tests based on simple-query branch). For this reason I'm keeping target branch as ``simple-query``

File: test/README.md [hunk 1]
```diff
@@ -1 +1 @@
-These tests are taken from repository https://github.com/datastax/nodejs-driver/tree/master
+Tests in ``./integration``, ``./other`` and ``./unit-not-supported`` are taken from repository https://github.com/datastax/nodejs-driver/tree/master
```
Comment (wprzytula) [inline test/README.md:1]: Are those double backticks on purpose? Why?

File: .github/workflows/tests.yml [hunk 2]
```diff
@@ -61,6 +61,8 @@ jobs:
           SCYLLA_URI=172.42.0.2:9042 node runner.js
       - name: Run tests
         run: npm test
+      - name: Run not fully ingegrated unit tests
```
Comment (wprzytula) [inline .github/workflows/tests.yml:64]: ingegrated -> integrated

File: test/unit-breaking/README.md [hunk 3]
```diff
@@ -0,0 +1 @@
+Direcory for unit tests that are activly not working with the current chanes
```
Comment (wprzytula) [inline test/unit-breaking/README.md:1]: Fixed typos & refactored:

`Directory for unit tests that are not currently not working with the current state of the driver. `

Comment (wprzytula) [reply test/unit-breaking/README.md:1]: Why `unit-breaking`? Maybe `unit-broken` or `unit-nonworking`?

**Review by wprzytula — CHANGES_REQUESTED** [review]

File: test/README.md (hunk 1 above)
Comment (adespawn) [reply test/README.md:1]: that's MD file, so marks given folders as code in markdown. Similar approach as in https://github.com/scylladb-zpp-2024-javascript-driver/scylladb-javascript-driver/blob/a9165c6a3d0b663bb0b2587f1d8fe81ea5a86b94/examples/README.md?plain=1#L1-L5

**Review by wprzytula — COMMENTED** [review]
Some checks are failing - why?

**Comment by adespawn** [comment]
> Some checks are failing - why?

The target branch currently doesn't have the required functionality to pass examples

**Review by wprzytula — APPROVED** [review]

File: test/unit-not-supported/geometry/line-string-tests.js
Comment (ZuzaOsa) [inline test/unit-not-supported/geometry/line-string-tests.js:1]: Geometry and graph tests are already testing new driver logic, they should be in working category (maybe renamed).

**Review by ZuzaOsa — CHANGES_REQUESTED** [review]

File: test/unit-not-supported/geometry/line-string-tests.js
Comment (wprzytula) [reply test/unit-not-supported/geometry/line-string-tests.js:1]: Shouldn't they be simply removed?

Comment (adespawn) [reply test/unit-not-supported/geometry/line-string-tests.js:1]: Those are the tests that check if the error is correctly thrown in those not supported features:
https://github.com/scylladb-zpp-2024-javascript-driver/scylladb-javascript-driver/pull/34#discussion_r1814283947

File: test/README.md
Comment (ZuzaOsa) [inline test/README.md:1]: Aren't tests in unit-broken also taken from the datastax repo?

File: test/unit-broken/README.md
Comment (ZuzaOsa) [inline test/unit-broken/README.md:1]: "Directory for unit tests that are not working with the current state of the driver. " instead

**Review by ZuzaOsa — CHANGES_REQUESTED** [review]

File: test/unit/README.md [hunk 4]
```diff
@@ -0,0 +1,2 @@
+These directory contains test, that check new driver logic and are expected to pass.
```
Comment (ZuzaOsa) [inline test/unit/README.md:1]: "contains tests" instead "contains test", "This directory" instead "These directory"

**Review by ZuzaOsa — CHANGES_REQUESTED** [review]

**Review by ZuzaOsa — APPROVED** [review]

**Pull request merged**